iroha_data_model/src/lib.rs

```diff
@@ -1655,6 +1655,19 @@ pub mod domain {
                 ))
             }
         }
+
+        /// Domain constructor with presetup accounts. Useful for testing purposes.
+        pub fn with_accounts(name: &str, accounts: impl IntoIterator<Item = Account>) -> Self {
+            let mut accounts_map = AccountsMap::new();
+            for account in accounts {
+                drop(accounts_map.insert(account.id.clone(), account));
```

Comment:
May be we can discuss somehow `unused_results` lint? I dont feel like we benefiting from it anyhow. Even worse this drop here is not semantically relevant (as drop for mutexguards for example). Besides this drives reader away. Main thing here is that `accounts_map` is getting inserted, not that we drop previous item from where.